### PR TITLE
Refactor storages

### PR DESCRIPTION
**Pull request by vladfedoriuk**
### Added
- Moved storages to a separate python package
- Made the storages independent of the rest of the PySDM codebase
- Provided tests for the storages
- Provided typings and docstrings

### Breaking changes

- For the indexed storages, created a new method `indexed_and_empty` instead of `emtpy` to resolve the issues with method signatures inconsistency - the signatures used to be different from the base Storage class.
- Moved some of the untility functions to the base Storage class.
- Renamed the methods for creating indexes, indexed, and pairwise storages.

### CODE REVIEW

**Comment by slayoo** [comment]
thank you @vladfedoriuk !!!

**Comment by slayoo** [comment]
@vladfedoriuk (CC: @abulenok, @piotrbartman) just to clarify, we are currently aiming at merging a stream of droplet-breakup PRs due to upcoming paper-revision deadline, and after getting that done, we will prioritize this PR. Thanks again for coming up with this refactor!

File: PySDM/storages/common/backend.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+
+class Backend(Protocol):
+    """
+    The backend base class.
+
+    The backend base class is used to define the interface of the backends.
+    """
+
+    ...
```
Comment (abulenok) [inline PySDM/storages/common/backend.py:24]: It seems to me that there is no point in having a `Protocol` class here. The `Backend` class doesn't possess any attributes or methods, so any class will be recognized as `Backend` by static type checkers.

File: PySDM/backends/numba.py [hunk 2]
```diff
@@ -7,42 +7,39 @@
...
     TerminalVelocityMethods,
+    NumbaStorageHolder,
 ):
-    Storage = ImportedStorage
-    Random = ImportedRandom
-
     default_croupier = "local"
 
+    # TODO #1185  https://github.com/pylint-dev/pylint/issues/9225
+    # pylint: disable=super-init-not-called
```
Comment (abulenok) [inline PySDM/backends/numba.py:37]: This pragma should not be needed here https://github.com/open-atmos/PySDM/issues/1185

**Pull request closed without merging**